### PR TITLE
Refactor Graphql Query Structure

### PR DESCRIPTION
**Pull request by nilsjung**

### CODE REVIEW

File: backend/graphql/databaseQuery.js [hunk 1]
```diff
@@ -2,65 +2,45 @@ const ConnectionData = require('../model/data');
...
-
-/**
- * the actual calculation just reduces the result to an square with radius on longitude and latitude axis.
- * To improve to filter per distance, you have to do some math here.
- * This solution should fit for now
- */
-async function getConnectionDataByRadius({ currentLocation, radius }) {
-    return new Promise((resolve, reject) => {
-        ConnectionData.find({
+    if (radius !== 0) {
```
Comment (karsil) [inline backend/graphql/databaseQuery.js]: Check for larger 0.

File: frontend/components/ButtonField/ButtonField.js [hunk 2]
```diff
@@ -25,13 +25,13 @@ class ButtonField extends React.Component {
...
                         style={buttonFieldStyles.button}
-                        onPress={() => this.props.getConnectionInfo()}
+                        onPress={() => this.props.getConnectionDataAction()}
                         title="Get All Data"
                     />
                     <Button
                         style={buttonFieldStyles.button}
                         onPress={() =>
-                            this.props.getConnectionInfoByRadius(this.props)
+                            this.props.getConnectionDataAction(50000)
```
Comment (karsil) [inline frontend/components/ButtonField/ButtonField.js]: Might declare radius variable more prominent.

File: frontend/components/Root/Root.Action.js [hunk 3]
```diff
@@ -44,10 +44,10 @@ export const requestLocation = () => {
         return getCurrentLocationLatLong()
             .then((location) => {
                 dispatch(updateGPS(location));
-                dispatch(getAllConnectionDataAction());
+                dispatch(getConnectionDataAction());
                 dispatch(isSearchingForLocation(false));
             })
-            .catch((err) => isSearchingForLocation(false));
+            .catch(() => isSearchingForLocation(false));
```
Comment (karsil) [inline frontend/components/Root/Root.Action.js]: dispatch?

File: frontend/graphql/mutation.js [hunk 4]
```diff
@@ -1,29 +1,23 @@
...
-    }
-`;
+function submitConnectionData({
+    location: { latitude, longitude },
+    provider,
+    platform,
+    connectionType,
+}) {
+    return client.mutate({
+        refetchQueries: [{ query: getAllConnectionData }],
```
Comment (karsil) [inline frontend/graphql/mutation.js]: **Throws error**: `You must wrap the query string in a "gql" tag`.
Has been checked by logging `createConnectionData`

**Review by karsil — CHANGES_REQUESTED** [review]

**Comment by karsil** [comment]
Very good call for this merge request. Much easier for future implementations!

File: backend/graphql/databaseQuery.js (hunk 1 above)
Comment (nilsjung) [reply backend/graphql/databaseQuery.js]: good point

File: frontend/components/Root/Root.Action.js (hunk 3 above)
Comment (nilsjung) [reply frontend/components/Root/Root.Action.js]: i will change this, although the error seems to be a bit older :grimacing:

File: frontend/graphql/mutation.js (hunk 4 above)
Comment (nilsjung) [reply frontend/graphql/mutation.js]: i will take a look at this.

**Review by karsil — APPROVED** [review]
Looks nice and is running!

**Pull request merged**